cartography/intel/aws/cloudfront.py

```diff
@@ -0,0 +1,201 @@
+        )
+
+
+@timeit
+def load_cloudfront_distributions(
+        neo4j_session: neo4j.Session,
+        data: List[Dict],
+        current_aws_account_id: str,
+        aws_update_tag: int,
+) -> None:
+    ingest_distribution = """
+        MERGE (d:CloudFrontDistribution{id: $distribution.ARN})
+        ON CREATE SET d.firstseen = timestamp()
+        SET d.arn = $distribution.ARN,
+            d.comment = $distribution.Comment,
+            d.domain_name = $distribution.DomainName,
+            d.enabled = $distribution.Enabled,
+            d.http_version = $distribution.HttpVersion,
+            d.is_ipv6_enabled = $distribution.IsIPV6Enabled,
+            d.last_modified_time = $distribution.LastModifiedTime,
+            d.status = $distribution.Status,
+            d.web_acl_id = $distribution.WebACLId,
+            d.geo_restriction_type = $distribution.Restrictions.GeoRestriction.RestrictionType,
+            d.geo_restriction_quantity = $distribution.Restrictions.GeoRestriction.Quantity,
+            d.geo_restriction_items = $distribution.Restrictions.GeoRestriction.Items,
+            d.aliases_quantity = $distribution.Aliases.Quantity,
+            d.aliases_items = $distribution.Aliases.Items,
+            d.origins_quantity = $distribution.Origins.Quantity,
+            d.origin_groups = $distribution.OriginGroups,
+            d.cache_behaviors = $distribution.CacheBehaviors,
+            d.default_cache_behavior = $distribution.DefaultCacheBehavior,
+            d.custom_error_responses = $distribution.CustomErrorResponses,
+            d.viewer_certificate = $distribution.ViewerCertificate,
+            d.lastupdated = $aws_update_tag
+        WITH d
+        MATCH (owner:AWSAccount{id: $AWS_ACCOUNT_ID})
+        MERGE (owner)-[r:RESOURCE]->(d)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = $aws_update_tag
+    """
+
+    for distribution in data:
+        distribution["LastModifiedTime"] = dict_date_to_epoch(distribution, 'LastModifiedTime')
+
+        if distribution["Restrictions"]["GeoRestriction"]["Quantity"] == 0:
+            distribution["Restrictions"]["GeoRestriction"]["Items"] = []
+
+        if distribution["Aliases"]["Quantity"] == 0:
+            distribution["Aliases"]["Items"] = []
+
+        distribution["OriginGroups"] = json.dumps(distribution["OriginGroups"])
+        distribution["CacheBehaviors"] = json.dumps(distribution["CacheBehaviors"])
+        distribution["DefaultCacheBehavior"] = json.dumps(distribution["DefaultCacheBehavior"])
+        distribution["CustomErrorResponses"] = json.dumps(distribution["CustomErrorResponses"])
+        distribution["ViewerCertificate"] = json.dumps(distribution["ViewerCertificate"])
+
+        neo4j_session.run(
+            ingest_distribution,
+            distribution=distribution,
+            AWS_ACCOUNT_ID=current_aws_account_id,
+            aws_update_tag=aws_update_tag,
+        )
+
+        load_distribution_origin(
```

Comment:
Try to load the distribution origins in a similar batch manner. You can load them in the same query as ingest_distribution.